CMakeLists.txt

```diff
@@ -28,14 +40,30 @@
 #                        ISOLATION_LEVEL 2
 #                        REGRESSION
 #                        BL2 True
-#                        BUILD_PROFILE profile_small)
+#                        BUILD_PROFILE profile_small
+#                        TFM_PARTITION_PROTECTED_STORAGE ON
+#                        TFM_PARTITION_INTERNAL_TRUSTE_STORAGE ON
+#                        TFM_PARTITION_CRYPTO ON
+#                        TFM_PARTITION_INITIAL_ATTESTATION ON
+#                        TFM_PARTITION_PLATFORM ON
+#                        TFM_PARTITION_AUDIT_LOG ON)
 function(trusted_firmware_build)
   set(options IPC REGRESSION)
-  set(oneValueArgs BINARY_DIR BOARD BL2 ISOLATION_LEVEL CMAKE_BUILD_TYPE BUILD_PROFILE)
+  set(oneValueArgs BINARY_DIR BOARD BL2 ISOLATION_LEVEL CMAKE_BUILD_TYPE
+    BUILD_PROFILE ${tfm_partitions})
```

Comment:
I suggest to use multi value argument here, like:
```
  set(oneValueArgs BINARY_DIR BOARD BL2 ISOLATION_LEVEL CMAKE_BUILD_TYPE BUILD_PROFILE)
  set(multiValueArgs PARTITIONS)
  cmake_parse_arguments(TFM "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
```

then caller can call this function as:
```
 trusted_firmware_build(.... PARTITIONS TFM_PARTITION_PROTECTED_STORAGE TFM_PARTITION_INTERNAL_TRUSTE_STORAGE TFM_PARTITION_INITIAL_ATTESTATION ...)
```

for the partitions to enable.

Reply:
We have to explicitly state `OFF` for all the partitions, so we need some more logic in the function in this case, but I agree that it is better seen from the callers perspective.

Reply:
you already have a `foreach` so would be very simple to adjust to something like:
```
set(partitions_off ${TFM_VALID_PARTITIONS})  # Defaulting everything to off.
list(REMOVE_ITEM partitions_off ${TFM_PARTITIONS}) # Those are provided by the user in the call, so remove from the off list

foreach(partition ${partitions_off})
    list(APPEND TFM_PARTITIONS_ARGS -D${partition}=OFF)
endforeach()

foreach(partition ${TFM_PARTITIONS})
    list(APPEND TFM_PARTITIONS_ARGS -D${partition}=ON)
endforeach()
```

Reply:
and then you can of course remove:
```
 foreach(partition ${TFM_VALID_PARTITIONS})
    list(FIND TFM_ENABLED_PARTITIONS ${partition} idx)
    if (idx EQUAL -1)
      set(val "OFF")
    else()
      set(val "ON")
    endif()
    list(APPEND TFM_PARTITIONS_ARGS -D${partition}=${val})
  endforeach()
```

Reply:
never mind.
I see now that I was already looking at the updated code.